plugins/auth-backend/src/providers/cloudflare-access/provider.ts

```diff
@@ -0,0 +1,255 @@
+/*
+ * Copyright 2022 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import {
+  AuthHandler,
+  AuthProviderRouteHandlers,
+  AuthResolverContext,
+  AuthResponse,
+  SignInResolver,
+} from '../types';
+import express from 'express';
+import * as _ from 'lodash';
+import { jwtVerify, createRemoteJWKSet } from 'jose';
+import { Profile as PassportProfile } from 'passport';
+import { AuthenticationError } from '@backstage/errors';
+import { createAuthProviderIntegration } from '../createAuthProviderIntegration';
+import { prepareBackstageIdentityResponse } from '../prepareBackstageIdentityResponse';
+import { makeProfileInfo } from '../../lib/passport';
+
+// JWT Web Token definitions are in the URL below
+// https://developers.cloudflare.com/cloudflare-one/identity/users/validating-json/
+export const CF_JWT_HEADER = 'cf-access-jwt-assertion';
+export const CF_AUTH_IDENTITY = 'cf-access-authenticated-user-email';
+
+/** @public */
+export type Options = {
+  /**
+   * Access team name
+   *
+   * When you configure Access, the public certificates are available at this
+   * URL, where your-team-name is your team name:
+   * https://<your-team-name>.cloudflareaccess.com/cdn-cgi/access/certs
+   */
+  teamName: string;
+  authHandler: AuthHandler<CloudflareAccessResult>;
+  signInResolver: SignInResolver<CloudflareAccessResult>;
+  resolverContext: AuthResolverContext;
+};
+
+/** @public */
+export type CloudflareAccessClaims = {
+  /**
+   * `aud` identifies the application to which the JWT is issued.
+   */
+  aud: string[];
+  /**
+   * `email` contains the email address of the authenticated user.
+   */
+  email: string;
+  /**
+   * iat and exp are the issuance and expiration timestamps.
+   */
+  exp: number;
+  iat: number;
+  /**
+   * `nonce` is the session identifier.
+   */
+  nonce: string;
+  /**
+   * `identity_nonce` is available in the Application Token and can be used to
+   * query all group membership for a given user.
+   */
+  identity_nonce: string;
+  /**
+   * `sub` contains the identifier of the authenticated user.
+   */
+  sub: string;
+  /**
+   * `iss` the issuer is the application’s Cloudflare Access Domain URL.
+   */
+  iss: string;
+  /**
+   * `custom` contains SAML attributes in the Application Token specified by an
+   * administrator in the identity provider configuration.
+   */
+  custom: string;
+};
+
+/** @public */
```

Comment:
this seems to be the only one that's _actually_ public all of the way to the external interface (as per the api-report above) - the other types should probably lose their `@public` if this is intended.